app/jobs/application_job.rb

```diff
@@ -5,4 +5,7 @@ class ApplicationJob < ActiveJob::Base
 
   # Most jobs are safe to ignore if the underlying records are no longer available
   # discard_on ActiveJob::DeserializationError
+
+  # We use good_job's concurrency features to cancel "extra" or duplicative runs of the same job
+  discard_on GoodJob::ActiveJobExtensions::Concurrency::ConcurrencyExceededError
```

Comment:
A few other options here:

1. Don't add this to the "base" job, only add it to the reports that use it (which is **26**)
2. Update `BaseReport` (which gets *most* of the reports), but misses a few
3. Make a ReportUniquenes mixin, that also DRY's up this bit in a bunch of different classes:

```ruby
  good_job_control_concurrency_with(
    enqueue_limit: 1,
    perform_limit: 1,
    key: -> { "#{REPORT_NAME}-#{arguments.first}" },
  )
```

pros: DRY
cons: implicit assumption first argument is always a date 😬

Reply:
Option 2 seems like the _correct_ way to do it. I worry that add this in here now plants a landmine for future users of this subclasses that expect the default behavior.

Reply:
I like 2? My worry is we eventually are doing concurrency control on other jobs, and we would want to see those errors, but maybe that's worrying about a problem that doesn't exist yet.

Reply:
Ok! Updated in 7554c88283f9e1d6ff3136a5e6422ddcad08206f to move to `Reports::BaseReport`. I added it directly to the non-BaseReport report-like jobs because BaseReport has a lot of methods that we don't use